build/features.py

```diff
@@ -994,180 +1000,85 @@ def configure(self, build, conf):
             # build.env.Append(LINKFLAGS = '/OPT:NOWIN98')
             # ALBERT: NOWIN98 is not supported in MSVC 2010.
 
-            if build.build_is_debug:
-                self.status = "Disabled (debug build)"
-            
             # http://msdn.microsoft.com/en-us/library/59a3b321.aspx
             # In general, you should pick /O2 over /Ox
-            elif optimize_level >= 1:
-                self.status = "Enabled -- Maximize Speed (/O2)"
-                build.env.Append(CCFLAGS='/O2')
-            # elif optimize_level >= 2:
-            #    self.status = "Enabled -- Maximum Optimizations (/Ox)"
-            #    build.env.Append(CCFLAGS = '/Ox')
+            build.env.Append(CCFLAGS='/O2')
+
+            # Historicaly our release packages are built with optimize=9.
+            if optimize_level == 'portable' or optimize_level == '' or optimize_level == 9:
+                # portable-binary: sse2 CPU (>= Pentium 4)
+                self.status = "portable: sse2 CPU (>= Pentium 4)"
+                build.env.Append(CCFLAGS='/arch:SSE2')
+                build.env.Append(CPPDEFINES=['__SSE__', '__SSE2__'])
+            elif optimize_level == 'native':
+                self.status = "native: exclusive for this CPU (%s)" % build.machine
+                build.env.Append(CCFLAGS='/favor:' + build.machine)
+            elif optimize_level == 'legacy':
+                self.status = "legacy: pure i386 code"
+            else:
+                raise Exception("optimize={} is not supported. "
```

Comment:
This will throw exceptions if someone uses the numbers. Could you please check for the legacy numbers above and map them to something sensible?